homeassistant/components/litterrobot/switch.py

```diff
@@ -0,0 +1,68 @@
+"""Support for Litter-Robot switches."""
+from homeassistant.helpers.entity import ToggleEntity
+
+from .const import DOMAIN
+from .hub import LitterRobotEntity
+
+
+class LitterRobotNightLightSwitch(LitterRobotEntity, ToggleEntity):
+    """Litter-Robot Night Light Switch."""
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self.robot.night_light_active
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return "mdi:lightbulb-on" if self.is_on else "mdi:lightbulb-off"
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the switch on."""
+        await self.perform_action_and_refresh(self.robot.set_night_light, True)
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the switch off."""
+        await self.perform_action_and_refresh(self.robot.set_night_light, False)
+
```

Comment:
This should be on the `light platform`

Reply:
Activating this mode on the LitterRobot actually doesn't directly turn the nightlight on or off. It just sets the "mode" of the nightlight - either "off" or something similar to "auto".

The actual device turns itself on or off based on occupancy and illumination when it's enabled. Looking at the state of the mode actually doesn't tell you the true state of the light. Which is why Nate made it a switch.

Reply:
As @joelmoses stated, it's not a true light, but a mode that is enabled.

Reply:
@joelmoses, since you have a unit and as a potential user of this integration, would it make more sense to title this entity/class "Auto Night Light" or "Night Light Mode" rather than simply "Night Light"?

Reply:
@natekspencer Maybe it'd be better to name the switch "Nightlight Mode" or something? :)

Reply:
That makes more sense to me.  

Without the context it definitely seems like it's directly controlling the state of a light.

Reply:
@natekspencer Crossed paths there!
Yes, I think "Night Light Mode" would be a good way to indicate the actual function.

Reply:
Yeah, let me rename it so it is a little more clear. I attempted to reproduce the Litter-Robot app's verbiage so it made sense to users coming from the companion app to HA. In the original app, it is called "Night Light", but in the autopets app, they renamed it to "Night Mode".